src/rgw/rgw_data_sync.cc

```diff
@@ -3200,15 +3230,17 @@ class InitBucketFullSyncStatusCR : public RGWCoroutine {
         }
       }
 
-      status.shards_done_with_gen.resize(num_shards);
+      status.shards_done_with_gen.resize(marker_mgr.get().size());
       status.incremental_gen = info.latest_gen;
 
       ldout(cct, 20) << "writing bucket sync status during init. state=" << status.state << ". marker=" << status.full.position.to_str() << dendl;
 
       // write bucket sync status
       using CR = RGWSimpleRadosWriteCR<rgw_bucket_sync_status>;
-      yield call(new CR(dpp, sync_env->async_rados, sync_env->svc->sysobj,
-                        status_obj, status, &objv, false));
+      yield {
+	call(new CR(dpp, sync_env->async_rados, sync_env->svc->sysobj,
+		    status_obj, status, &objv, false));
+      }
```

Comment:
why change this?

Reply:
Leftover part of a change that didn't get made.